sonic_platform_base/module_base.py

```diff
@@ -196,6 +206,62 @@ def get_maximum_consumed_power(self):
         """
         raise NotImplementedError
 
+    ##############################################
+    # SmartSwitch methods
+    ##############################################
+
+    def get_reboot_cause(self):
+        """
+        Retrieves the DPU ID. Returns None for non-smartswitch chassis.
+
+        Returns:
+            An integer, indicating the DPU ID. DPU0 returns 1, DPUX returns X+1
+            Returns '0' on switch module
+        """
+        raise NotImplementedError
+
+    def get_reboot_cause(self):
+        """
+        Retrieves the cause of the previous reboot of the DPU module
+
+        Returns:
+            A tuple (string, string) where the first element is a string
+            containing the cause of the previous reboot. This string must
+            be one of the predefined strings in this class. If the first
+            string is "REBOOT_CAUSE_HARDWARE_OTHER", the second string can be
+            used to pass a description of the reboot cause.
+
+            Some more causes are appended to the existing list to handle other
+            modules such as DPUs.
+            Ex: REBOOT_CAUSE_POWER_LOSS, REBOOT_CAUSE_HOST_RESET_DPU,
+            REBOOT_CAUSE_HOST_POWERCYCLED_DPU, REBOOT_CAUSE_SW_THERMAL,
+            REBOOT_CAUSE_DPU_SELF_REBOOT
+        """
+        raise NotImplementedError
+
+    def get_state_info(self):
+        """
+        Retrieves the dpu state object having the detailed dpu state progression.
+        Fetched from ChassisStateDB.
+
+        Returns:
+            An object instance of the DPU_STATE (see DB schema)
+            Returns None on switch module
+        """
+        raise NotImplementedError
+
+    def get_health_info(self):
```

Comment:
The mechanism of fetching the health information from the DPU should be vendor-independent. Do we need to add it to the Platform API?

Reply:
Yes, though the fetching of the DPU health-info is vendor-independent, as per the pmon hld that we all agreed upon the health-info needs to be updated into redis chassisStateDB and its contents is exactly the same as the Switch health-info. So, in a way the health-info structure is fixed besides the vendor specific fields. There are some emerging use-cases where this will come in handy. So let us keep it.

Reply:
All of us have agreed to remove the "get_health_info" API and it is removed in the HLD